server/service/osquery.go

```diff
@@ -1083,6 +1083,44 @@ func (svc *Service) SubmitDistributedQueryResults(
 	return nil
 }
 
+// globalPolicyAutomationsEnabled returns true if any of the global policy automations are enabled.
+// globalPolicyAutomationsEnabled and teamPolicyAutomationsEnabled are effectively identical.
+// We could not use Go generics because Go generics does not support accessing common struct fields right now.
+// The umbrella Go issue tracking this: https://github.com/golang/go/issues/63940
+func globalPolicyAutomationsEnabled(webhookSettings fleet.WebhookSettings, integrations fleet.Integrations) bool {
+	if webhookSettings.FailingPoliciesWebhook.Enable {
+		return true
+	}
+	for _, j := range integrations.Jira {
+		if j.EnableFailingPolicies {
+			return true
+		}
+	}
+	for _, z := range integrations.Zendesk {
+		if z.EnableFailingPolicies {
+			return true
+		}
+	}
+	return false
+}
+
+func teamPolicyAutomationsEnabled(webhookSettings fleet.TeamWebhookSettings, integrations fleet.TeamIntegrations) bool {
+	if webhookSettings.FailingPoliciesWebhook.Enable {
+		return true
+	}
+	for _, j := range integrations.Jira {
+		if j.EnableFailingPolicies {
+			return true
+		}
+	}
+	for _, z := range integrations.Zendesk {
+		if z.EnableFailingPolicies {
+			return true
+		}
+	}
+	return false
+}
+
```

Comment:
Exact same code here: https://github.com/fleetdm/fleet/blob/80c574298e73210d1b2cd467c790b8a2494a0eb6/server/policies/failing_policies.go#L203-L225

Should we add this method to the `server/fleet` package?

Reply:
Yes, similar. The logic seemed so simple that I didn't think it was worth refactoring to combine these two functionalities.